File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
    const dfs = (idx, sum) => {
        if (idx >= numbers.length) {
            if (target === sum) answer++;
            return;
        }
        dfs(idx + 1, sum + numbers[idx]);
        dfs(idx + 1, sum - numbers[idx]);
    };

    dfs(0, 0);
    return answer;
}

File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;

int dfs(const vector<int>& numbers, const int& target, int order, int res, int answer) {
    if (order >= numbers.size())
        return (res == target) ? answer + 1 : answer;

    answer = dfs(numbers, target, order + 1, res + numbers[order], answer);
    answer = dfs(numbers, target, order + 1, res - numbers[order], answer);
    
    return answer;
}

int solution(vector<int> numbers, int target) {
    return dfs(numbers, target, 0, 0, 0);
}